### PR TITLE
support  sdk5.1.0 Validation functions

### PR DESCRIPTION
**Pull request by matemax**
support  sdk5.1.0 
use Validation functions
update liveness

### CODE REVIEW

File: lunavl/sdk/detectors/facedetector.py [hunk 1]
```diff
@@ -307,6 +299,31 @@ def redetectOne(  # noqa: F811
             return FaceDetection(detectRes, image)
         return None
 
+    def _validateBatchReDetectInput(self, coreImages: List[CoreImage], detectAreas: List[List[Detection]]):
```
Comment (gluhar2006) [inline lunavl/sdk/detectors/facedetector.py]: comments

File: lunavl/sdk/detectors/facedetector.py [hunk 2]
```diff
@@ -307,6 +299,31 @@ def redetectOne(  # noqa: F811
...
+            validationError, imagesErrors = self._detector.validate([coreImages], [[detectAreas]])
+        if validationError.isOk:
+            return
+        if validationError.error != FSDKError.ValidationFailed:
+            raise LunaSDKException(LunaVLError.fromSDKError(validationError), imagesErrors)
+        if not isinstance(coreImages, list):
+            raise LunaSDKException(LunaVLError.fromSDKError(imagesErrors[0]))
+        errors = []
+
+        for imageErrors in imagesErrors:
```
Comment (gluhar2006) [inline lunavl/sdk/detectors/facedetector.py:325]: looks the as for detectors/base.py errors processing...dry?

File: lunavl/sdk/detectors/base.py [hunk 3]
```diff
@@ -134,34 +132,43 @@ def getArgsForCoreRedetect(images: List[ImageForRedetection]) -> Tuple[List[Core
...
+        detector: core face or body detector
         coreImages: list of core images
         detectAreas: list of detect areas for core images
-        getErrorFunction: function to collect error by
     Raises:
-        LunaSDKException(LunaVLError.BatchedInternalError) with collected errors in context
+        LunaSDKException: if validation are failed or data is not valid
     """
+    if not isinstance(coreImages, list):
+        validationError, imageErrors = detector.validate([coreImages], [detectAreas], 1)
```
Comment (gluhar2006) [inline lunavl/sdk/detectors/base.py]: use keyword for `1`

File: lunavl/sdk/estimators/face_estimators/livenessv1.py [hunk 4]
```diff
@@ -84,24 +108,19 @@ def quality(self) -> float:
 class LivenessV1Estimator(BaseEstimator):
     """
     Liveness estimator version 1 (LivenessOneShotRGBEstimator).
-
-    Attributes:
-        principalAxes: maximum value of Yaw, pitch and roll angles for estimation
     """
 
     __slots__ = ("principalAxes",)
```
Comment (gluhar2006) [inline lunavl/sdk/estimators/face_estimators/livenessv1.py]: looks superfluous

File: lunavl/sdk/detectors/facedetector.py (hunk 2 above)
Comment (matemax) [reply lunavl/sdk/detectors/facedetector.py:325]: for human detector has not validation function

File: lunavl/sdk/detectors/base.py (hunk 3 above)
Comment (matemax) [reply lunavl/sdk/detectors/base.py]: yes... and no=(
image

Comment (gluhar2006) [reply lunavl/sdk/detectors/base.py]: smth like this?
```
limit = 1
validationError, imageErrors = detector.validate([coreImages], [detectAreas], limit)
```

File: lunavl/sdk/detectors/facedetector.py (hunk 1 above)
Comment (matemax) [reply lunavl/sdk/detectors/facedetector.py]: +

File: lunavl/sdk/estimators/face_estimators/livenessv1.py (hunk 4 above)
Comment (matemax) [reply lunavl/sdk/estimators/face_estimators/livenessv1.py]: +

File: lunavl/sdk/detectors/base.py (hunk 3 above)
Comment (matemax) [reply lunavl/sdk/detectors/base.py]: +

**Pull request merged**